src/MessageRepositoryTestTooling/MessageRepositoryTestCase.php

```diff
@@ -134,4 +195,17 @@ public function failing_to_retrieve_messages_after_version(): void
 
         $repository->retrieveAllAfterVersion($this->aggregateRootId, 5);
     }
+
+    /**
+     * @test
+     */
+    public function failing_to_paginate(): void
+    {
+        $this->tableName = 'invalid';
+        $repository = $this->messageRepository();
+
+        self::expectException(UnableToRetrieveMessages::class);
```

Comment:
Is this actually testing pagination failures (`failing to paginate` vs `unable to retrieve messages`)?

Reply:
To me pagination is just a way to retrieve messages. I'm quite ok with using this exception for it as the error circumstances are most likely the same, wrong table/schema or network failure.

Reply:
Agreed!